go/vt/topo/server.go

```diff
@@ -294,26 +318,20 @@ func (ts *Server) ConnForCell(ctx context.Context, cell string) (Conn, error) {
 	return conn, nil
 }
 
-// CellToRegionMapper function is a wrapper around topo.Server#GetRegionByCell with caching and error handling
-func (ts *Server) CellToRegionMapper() func(cell string) string {
+// GetRegionByCell returns the region group this `cell` belongs to, if there's none, it returns the `cell` as region.
+func GetRegionByCell(cell string) string {
 
-	memoize := make(map[string]string)
-	ctx := context.Background()
-
-	return func(cell string) string {
-		if ts == nil {
-			return cell
-		}
-		if region, ok := memoize[cell]; ok {
-			return region
-		}
-		if region, err := ts.GetRegionByCell(ctx, cell); err == nil {
-			memoize[cell] = region
-			return region
-		}
-		// for backward compatibility, when region isn't available, it's the same as given cell
-		return cell
+	region, _ := regions.cellsToRegions[cell]
```

Comment:
Is it safe to read without lock? Seems to me like that's the case, but just double checking.